src/core/main.c

```diff
@@ -1480,6 +1480,32 @@ int main(int argc, char *argv[]) {
                         (void) write_string_file("/proc/sys/kernel/core_pattern", "|/bin/false", 0);
         }
 
+        if (arg_system) {
+                /* When started as PID1, the kernel uses /dev/console
+                 * for our stdios and uses TERM=linux whatever the
+                 * backend device used by the console. We try to make
+                 * a better guess here since some consoles might not
+                 * have support for color mode for example.
+                 *
+                 * However if TERM was configured through the kernel
+                 * command line then leave it alone.
+                 *
+                 * We expect TERM to be set correctly if run inside a
+                 * container. */
+                if (detect_container() <= 0) {
+                        _cleanup_free_ char *term = NULL;
+
+                        if (get_proc_cmdline_key("TERM=", &term) == 0)
```

Comment:
if this function fails then the following strdup is likely to fail for the same reason: the system went out of memory. So I even not tried to continue in this case and the next condition "if (!term)" should catch this case.

But I can use a "<=" if you prefer.